proxy/proxy/httpserver.py

```diff
@@ -0,0 +1,171 @@
+"""proxy server"""
+from http import HTTPStatus
+from http.server import ThreadingHTTPServer, BaseHTTPRequestHandler
```

Comment:
From what I know `http.server` is not recommended for security reasons. Would it be possible to use another approach?

https://docs.python.org/3/library/http.server.html

Reply:
I looked into some other options but I couldn't find a good one to handle low level data for the proxy.

Reply:
what about a simple flask app, something like:

```python
from flask import Flask, request, Response
import requests

app = Flask(__name__)


@app.route('/', methods=['POST'])
def proxy_request():
    """Append a header and pass request on."""
    runtime_url = "https://runtime.quantum.ibm.com"
    gateway_url = "https://middleware.quantum.ibm.com"

    headers = dict(request.headers)
    data = dict(request.data)
    # add our header(s)
    headers['serverless-id'] = "12345"
    
    # Send request to runtime
    r = requests.post(runtime_url, headers=headers, data=data)
    # update our DB
    r2 = requests.post(gateway_url+"/update/jobconfig", ....)

    # return response to user
    resp = Response("request forwarded")
    resp.headers = r.headers
    resp.set_data(r.text)
    return resp


if __name__ == "__main__":
    app.run(debug=True, host='0.0.0.0', port=int(8080))
```

and since we need to use Istio anyways, in theory we could lean on [Istio's traffic management](https://istio.io/latest/docs/concepts/traffic-management/) to redirect user calls to runtime to our proxy service...

Not sure how practical this would be, just trying to help think of other options if `http.server` won't work...

Reply:
> couldn't find a good one to handle low level data for the proxy

What kind of low-level data was causing problems? While it seems like the security issues mentioned in the python docs don't apply in our case, the "`http.server` is not recommended for production" warning makes me a little uncomfortable...

Reply:
I believe that our use of this http.server is safe.  We handle all requests by ourself. So I guess it's OK.  The proxy implementation may not requires very low level but it requires header handling and raw data manipulation. Thanks!

Reply:
I think it's more likely than not that we're safe too, but I'm not as confident as I'd like...
@Tansito @IceKhan13 wdyt?

Reply:
Yeah, I agree with @psschwei here. If there was the only option to manage header handling and raw data manipulation I agree but I would like to take a look to other more "production ready" options.